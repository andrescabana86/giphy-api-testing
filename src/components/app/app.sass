.app
  align-items: center
  display: flex
  flex-direction: column
  &__search
    border-radius: 10px
    border: 1px solid lightgray
    display: flex
    margin: 8px
    padding: 6px 12px
    width: 480px