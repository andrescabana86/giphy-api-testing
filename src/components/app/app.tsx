import React from 'react'
import './app.sass'

export class App extends React.PureComponent {
  render() {
    return (
      <h1 data-testid="title" className="title">Welcome to my boilerplate</h1>
    )
  }
}
