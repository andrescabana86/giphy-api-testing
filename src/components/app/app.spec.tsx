import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { App } from '@components/app/app'
import { disableFetchMocks, enableFetchMocks } from 'jest-fetch-mock'

beforeAll(enableFetchMocks)
afterAll(disableFetchMocks)

describe('App', () => {
  afterEach(cleanup)

  it('should render App component', () => {
    // Arrange
    fetchMock.mockResponseOnce(JSON.stringify({ data: [] }))

    // Act
    const { asFragment } = render(<App />)

    // Assert
    expect(asFragment()).toMatchSnapshot()
  })
})
