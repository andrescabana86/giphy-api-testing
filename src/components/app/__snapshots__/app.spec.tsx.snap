// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App component 1`] = `
<DocumentFragment>
  <h1
    class="title"
    data-testid="title"
  >
    Welcome to my boilerplate
  </h1>
</DocumentFragment>
`;
